joomla/joomla-cms
# Be able to use unique language strings for the rules field

### Summary of Changes

Right now in Joomla core, the way to override a language string is to define the key in either the designated overrides file or load it in a file loaded after the version you want to override.  This is problematic in some cases because a key gets used in multiple contexts and if you were to load files in the wrong order this results in the wrong translation being displayed.  Additionally, this behavior makes it really difficult to create a process for caching a `JLanguage` instance with all of the language strings loaded into memory (you replace the processing time loading INIs into memory by just starting with the object having everything).

`JText` already exposes an API for using alternate language strings; it should be used in locations where we know localized contexts of a string may be created.  This PR does just that for the two keys in the rules form field that are commonly overridden by extensions. 

### Testing Instructions

In the com_content component config and article edit view, the permissions strings should use the localized keys for this component.

### Documentation Changes Required

N/A

### Language B/C Concerns

Though these keys are renamed in this PR, there is no B/C concern with translations.  `JText::alt()` checks if the alternate key (built by appending the second parameter to the first parameter) exists before using it, so if it doesn't exist, the base key (which will probably already be translated on a current 3.6 package) will be used.
